internal/pkg/workspace/workspace.go

```diff
@@ -60,16 +62,20 @@ const (
 // Summary is a description of what's associated with this workspace.
 type Summary struct {
 	Application string `yaml:"application"` // Name of the application.
-
-	Path string // absolute path to the summary file.
+	Path        string `yaml:"-"`           // absolute path to the summary file.
 }
 
 // Workspace typically represents a Git repository where the user has its infrastructure-as-code files as well as source files.
 type Workspace struct {
 	workingDirAbs string
 	copilotDirAbs string
-	fs            *afero.Afero
-	logger        func(format string, args ...interface{})
+
+	summary       *Summary
+	summaryErr    error
+	summarizeOnce sync.Once
```

Comment:
maybe a comment that these should only be accessed via `Summary()`, not directly

Reply:
Done!